src/Microsoft.DotNet.XHarness.TestRunners.Xunit/ThreadlessXunitTestRunner.cs

```diff
@@ -0,0 +1,137 @@
+{
+    internal class ThreadlessXunitTestRunner
+    {
+        public int Run(string assemblyFileName, bool printXml, IEnumerable<string> excludedTraits)
+        {
+            var filters = new XunitFilters();
+            foreach (var trait in excludedTraits ?? Array.Empty<string>())
+            {
+                ParseEqualSeparatedArgument(filters.ExcludedTraits, trait);
+            }
+
+            var configuration = new TestAssemblyConfiguration() { ShadowCopy = false, ParallelizeAssembly = false, ParallelizeTestCollections = false, MaxParallelThreads = 1 };
+            var discoveryOptions = TestFrameworkOptions.ForDiscovery(configuration);
+            var discoverySink = new TestDiscoverySink();
+            var diagnosticSink = new ConsoleDiagnosticMessageSink();
+            var testOptions = TestFrameworkOptions.ForExecution(configuration);
+            var testSink = new TestMessageSink();
+            var controller = new Xunit2(AppDomainSupport.Denied, new NullSourceInformationProvider(), assemblyFileName, configFileName: null, shadowCopy: false, shadowCopyFolder: null, diagnosticMessageSink: diagnosticSink, verifyTestAssemblyExists: false);
+
+            discoveryOptions.SetSynchronousMessageReporting(true);
+            testOptions.SetSynchronousMessageReporting(true);
+
+            Console.WriteLine($"Discovering tests for {assemblyFileName}");
+            var assembly = Assembly.LoadFrom(assemblyFileName);
+            var assemblyInfo = new global::Xunit.Sdk.ReflectionAssemblyInfo(assembly);
+            var discoverer = new ThreadlessXunitDiscoverer(assemblyInfo, new NullSourceInformationProvider(), discoverySink);
+
+            discoverer.FindWithoutThreads(includeSourceInformation: false, discoverySink, discoveryOptions);
+            discoverySink.Finished.WaitOne();
+            var testCasesToRun = discoverySink.TestCases.Where(filters.Filter).ToList();
+            Console.WriteLine($"Discovery finished.");
+
+            var summarySink = new DelegatingExecutionSummarySink(testSink, () => false, (completed, summary) => { Console.WriteLine($"Tests run: {summary.Total}, Errors: 0, Failures: {summary.Failed}, Skipped: {summary.Skipped}{Environment.NewLine}Time: {TimeSpan.FromSeconds((double)summary.Time).TotalSeconds}s"); });
+            var resultsXmlAssembly = new XElement("assembly");
+            var resultsSink = new DelegatingXmlCreationSink(summarySink, resultsXmlAssembly);
+
+            testSink.Execution.TestPassedEvent += args => { Console.WriteLine($"[PASS] {args.Message.Test.DisplayName}"); };
+            testSink.Execution.TestSkippedEvent += args => { Console.WriteLine($"[SKIP] {args.Message.Test.DisplayName}"); };
+            testSink.Execution.TestFailedEvent += args => { Console.WriteLine($"[FAIL] {args.Message.Test.DisplayName}{Environment.NewLine}{ExceptionUtility.CombineMessages(args.Message)}{Environment.NewLine}{ExceptionUtility.CombineStackTraces(args.Message)}"); };
```

Comment:
Lets move this to use the logger, which you'll pass from the app entry point. That way all runners have the same interface, and we can write the a log file or any other location if we want. For example, if we want to run this on an iOS device, we will need a tcp writer to the usb port, so better be ready now.